docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+
+|     |     |
+|-----|-----|
+|**Pros** | The program will be able to detect all category name inside the ingredientList arraylist. |
+|**Cons** | This method will be slow when facing a huge amount of data in the arraylist as the program may have to go through every single item in the arraylist. |
+
+Alternative 2 (current choice): Creating a fixed array which includes the order and all possible category names.
+
+|     |     |
+|-----|-----|
+|**Pros** | Users would be able to get the details of the particular recipe accurately and fast. |
+|**Cons** | Program will not be able to handle any ingredient which isn’t belonging to the category names in the fixed array.  |
+
+#### 4.1.3. Delete all/ specific ingredients(s)
+The deletion feature for ingredients allows the user to delete ingredients either by the name or index of the ingredients. In addition to that, it allows users to reduce the quantity of a specific ingredient. 
+
+![DeleteIngredient Sequence Diagram](images/deleteIngredientSequenceDiagram.png)
+
+##### Implementation
+When the user attempts to reduce the quantity of ingredient at index 1 of the ingredients inventory by 4,  the `Kitchen Helper`, ‘Parser’ and ‘DeleteRecipeCommand` class will be called upon. The following sequence of steps will then occur: 
+1. The user keyed in “deleteingredient /i 1 /q 4”`. 
+    2. A `UI` object will be created and it will call `UI#getUserCommand()` method to take in the input that the user has keyed in.
+    3. A `String` object will be returned and saved into the `userCommandInput` variable in `Kitchen Helper`.
+    4. The variable `userCommandInput` is being parsed into the `Parser` class as an argument for this method `Parser#parseUserCommand`.
+2. The command inserted by the user is being parsed into the `Parser` and a new `Command` object is being created. 
+    2. The variable `userCommandInput` will be identified as `deleteingredient` in the `Parser#parseUserCommand()`.The `Parser#prepareDeleteIngredient()` is being called to prepare the `userCommandInput` string to create a `DeleteIngredientCommand` object.
+    3. The `DeleteIngredientCommand` object is created with the ingredientIndex and quantity set to 4. 
+3. The Command is now being executed. 
+    2. The `DeleteIngredientCommand#execute()` will be called.
+    3. As this is a deletion by ingredient index, the `ingredientIndex` variable is not null. As the `ingredientIndex` is not null, `DeleteIngredientCommand#deleteIngredientByIndex()`.
+    4. Next, `DeleteIngredientCommand#deleteIngredient()` is called to reduce the quantity of this ingredient since the `quantity` variable is not null. 
+    6. Next, `DeleteIngredientCommand#updateNewQuantity()` will be called to update the quantity of this ingredient in our ingredients’ inventory.
+    7. Lastly, a String called `feedbackToUser`will be returned to the user to inform the user of the outcome of the command. 
```

Comment:
Would you like to consider splitting these chunk of information with use of blank space and varied bulleting? It could help if the visuals are integrated into your explanation for easier referral.